ssr/src/utils/event_streaming/events.rs

```diff
@@ -726,3 +733,153 @@ impl ProfileViewVideo {
         }
     }
 }
+
+#[derive(Default)]
+pub struct TokenCreationStarted;
+
+impl TokenCreationStarted {
+    pub fn send_event(
+        &self,
+        sns_init_payload: SnsInitPayload,
+        cans_store: RwSignal<Option<Canisters<true>>>,
+    ) {
+        #[cfg(all(feature = "hydrate", feature = "ga4"))]
+        {
+            let user = user_details_can_store_or_ret!(cans_store);
+            let details = user.details;
+
+            let user_id = details.principal;
+            let canister_id = user.canister_id;
+
+            // token_creation_started - analytics
+            send_event(
+                "token_creation_started",
+                &json!({
+                    "user_id": user_id,
+                    "canister_id": canister_id,
+                    "token_name": sns_init_payload.token_name,
+                    "token_symbol": sns_init_payload.token_symbol,
+                    "name": sns_init_payload.name
+                }),
+            );
+        }
+    }
+}
+
+#[derive(Default)]
+pub struct TokenCreationCompleted;
+
+impl TokenCreationCompleted {
+    pub fn send_event(
+        &self,
+        sns_init_payload: SnsInitPayload,
+        cans_store: RwSignal<Option<Canisters<true>>>,
+    ) {
+        #[cfg(all(feature = "hydrate", feature = "ga4"))]
+        {
+            let user = user_details_can_store_or_ret!(cans_store);
+            let details = user.details;
+
+            let user_id = details.principal;
+            let canister_id = user.canister_id;
+
+            // token_creation_completed - analytics
+            send_event(
+                "token_creation_completed",
+                &json!({
+                    "user_id": user_id,
```

Comment:
can we also send the token img url or base64 ?

Reply:
i added it as a base64 string, in the field "logo"